service/history/configs/task.go

```diff
@@ -64,6 +68,34 @@ func ConvertWeightsToDynamicConfigValue(
 	return weightsForDC
 }
 
+func ConvertDynamicConfigValueToWeights(
+	weightsFromDC map[string]interface{},
+) map[int]int {
+	weights := make(map[int]int)
+	for key, value := range weightsFromDC {
+		intKey, err := strconv.Atoi(strings.TrimSpace(key))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to convert dynamicconfig map value to int map: %v is not integer", key))
```

Comment:
if operator make a mistake, will this crush the service?

Reply:
and do we really need dynamic weight?

Reply:
Good point. let me emit a log and fallback to default weight if dc value can't be parsed.

I think dynamic config is still necessary as it's hard to estimate the real effect of those weights without looking at production metrics. Existing iwrr scheduler implementation don't support change the weight on the fly. I will modify it later to support it and make the weights real dynamic.

Reply:
>I will modify it later to support it and make the weights real dynamic.

before making everything dynamic, let us sync, we may need to evaluate "making everything dynamic all the time" vs "ability to reload according to signal"